src/backend/distributed/sql/udfs/run_command_on_all_nodes/latest.sql

```diff
@@ -0,0 +1,23 @@
+DROP FUNCTION IF EXISTS pg_catalog.run_command_on_all_nodes;
+
+CREATE FUNCTION pg_catalog.run_command_on_all_nodes(command text, parallel bool default true, OUT nodeid int, OUT success bool, OUT result text)
+	RETURNS SETOF record
+	LANGUAGE plpgsql
+	AS $function$
+DECLARE
+	nodenames text[];
+	ports int[];
+	commands text[];
+BEGIN
+	WITH citus_nodes AS (
+		SELECT * FROM pg_dist_node WHERE isactive = 't' ORDER BY nodename, nodeport
```

Comment:
should exclude nodes from different clusters (read replicas)

Reply:
(I wrote this comment 3rd time now, as Ozan pushes new changes it somehow gets lost:) )

Also probably filter inactive nodes as well because those might be inaccessible.


@halilozanakgul what happens if one of the nodes cannot be reached? `citus_dist_stat_activity` gives a warning but still returns the result from healthy nodes. I think that behavior is better (over throwing an error and not showing anything). Is it similar for this one?

Reply:
I saw the 2nd one 😄 I'm just thinking how to do it.

> @halilozanakgul what happens if one of the nodes cannot be reached?

`citus_stat_activity` filters with the success column from this function. So only the results of successful commands but no warning.

Reply:
> citus_stat_activity filters with the success column from this function. So only the results of successful commands but no warning.

How hard is it to print the warnings? Wouldn't be nice to warn the user?